imap_processing/tests/pointing_frame/test_pointing_frame_handler.py

```diff
@@ -0,0 +1,136 @@
+    """Create path to kernels."""
+
+    test_dir = (
+        Path(sys.modules[__name__.split(".")[0]].__file__).parent
+        / "tests"
+        / "pointing_frame"
+        / "test_data"
+    )
+
+    kernels = [
+        "de430.bsp",
+        "naif0012.tls",
+        "imap_science_0001.tf",
+        "imap_sclk_0000.tsc",
+        "imap_wkcp.tf",
+        "imap_spin.bc",
+    ]
+
+    for file in test_dir.iterdir():
+        if file.name in kernels:
+            shutil.copy(file, tmp_path / file.name)
+
+    return tmp_path
+
+
+@pytest.fixture()
+def create_kernel_list(kernel_path):
+    """Create kernel lists."""
+    kernels = [str(file) for file in kernel_path.iterdir()]
+    ck_kernel = [
+        str(file) for file in kernel_path.iterdir() if file.name == "imap_spin.bc"
+    ]
+
+    return kernels, ck_kernel
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_get_coverage(create_kernel_list):
+    """Tests get_coverage function."""
+    kernels, ck_kernel = create_kernel_list
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_coverage(str(ck_kernel[0]))
+
+    assert et_start == 802008069.184905
+    assert et_end == 802094467.184905
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_create_pointing_frame(monkeypatch, kernel_path, create_kernel_list):
+    """Tests create_pointing_frame function."""
+    monkeypatch.setenv("EFS_MOUNT_PATH", str(kernel_path))
+    _, ck_kernel = create_kernel_list
+    create_pointing_frame()
+
+    # After imap_dps.bc has been created.
+    kernels = [str(file) for file in kernel_path.iterdir()]
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_coverage(str(ck_kernel[0]))
+
+        rotation_matrix_1 = spice.pxform("ECLIPJ2000", "IMAP_DPS", et_start + 100)
+        rotation_matrix_2 = spice.pxform("ECLIPJ2000", "IMAP_DPS", et_start + 1000)
+
+    # All the rotation matrices should be the same.
+    assert np.array_equal(rotation_matrix_1, rotation_matrix_2)
+
+    # Nick Dutton's MATLAB code result
+    rotation_matrix_expected = np.array(
+        [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
+    )
+    np.testing.assert_allclose(rotation_matrix_1, rotation_matrix_expected, atol=1e-4)
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_z_axis(create_kernel_list):
+    """Tests Inertial z axis and provides visualization."""
+    kernels, ck_kernel = create_kernel_list
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_coverage(str(ck_kernel[0]))
+
+        # Converts rectangular coordinates to spherical coordinates.
+        q_avg, z_eclip_time = average_quaternions(et_times)
+        z_avg_expected = spice.q2m(list(q_avg))[:, 2]
+        _, z_avg = create_rotation_matrix(et_times)
+
+        assert np.array_equal(z_avg, z_avg_expected)
```

Comment:
Can this be moved into another test and this one removed. I generally think that plots are an algorithm validation step that you could save and add to a ticket rather than doing plotting in automated tests.

Reply:
I find it helpful to have it in one place but I'll do a @pytest.mark.skip